code/_globalvars/lists/flavor_misc.dm

```diff
@@ -40,6 +40,117 @@ GLOBAL_LIST_EMPTY(caps_list)
 GLOBAL_LIST_INIT(ghost_forms_with_directions_list, list("ghost")) //stores the ghost forms that support directional sprites
 GLOBAL_LIST_INIT(ghost_forms_with_accessories_list, list("ghost")) //stores the ghost forms that support hair and other such things
 
+GLOBAL_LIST_INIT(ai_core_display_screens, list(
+	":thinking:",
+	"Alien",
+	"Angel",
+	"Bliss",
+	"Blue",
+	"Clown",
+	"Database",
+	"Dorf",
+	"Firewall",
+	"Fuzzy",
+	"Gentoo",
+	"Glitchman",
+	"Goon",
+	"Green",
+	"Hades",
+	"Heartline",
+	"Helios",
+	"House",
+	"Inverted",
+	"Matrix",
+	"Monochrome",
+	"Murica",
+	"Nanotrasen",
+	"Not Malf",
+	"President",
+	"Random",
+	"Red",
+	"Red October",
+	"Static",
+	"Syndicat Meow",
+	"Text",
+	"Too Deep",
+	"Triumvirate",
+	"Triumvirate-M"))
+
+/proc/resolve_ai_icon(input)
+	if(!input)
+		return "ai"
+	else
+		if(input == "Random")
+			input = pick(GLOB.ai_core_display_screens - "Random")
+		switch(input)
+			if(":thinking:")
+				return "ai-:thinking:"

```

Comment:
If you rename a few of the iconstates you could collapse this entire switch into just `return "ai-[ckey(input)]"`